scipy/linalg/flapack_gen_tri.pyf.src

```diff
@@ -66,6 +66,32 @@ subroutine <prefix>gttrs(trans, n, nrhs, dl, d, du, du2, ipiv, b, ldb, info)
 end subroutine gttrs
 
 
+subroutine <prefix>gtcon(norm,n,dl,d,du,du2,ipiv,anorm,rcond,work,iwork,info)
+    ! ?GTCON estimates the reciprocal of the condition number of a real
+    ! tridiagonal matrix A using the LU factorization as computed by
+    ! ?GTTRF.
+    ! An estimate is obtained for norm(inv(A)), and the reciprocal of the
+    ! condition number is computed as RCOND = 1 / (ANORM * norm(inv(A))).
+    threadsafe
+    callstatement (*f2py_func)(norm,&n,dl,d,du,du2,ipiv,&anorm,&rcond,work,iwork,&info)
+    callprotoargument char*, F_INT*, <ctype>*, <ctype>*, <ctype>*, <ctype>*, F_INT*, <ctypereal>*, <ctypereal>*, <ctype>*, <F_INT, F_INT, float,double>*, F_INT*
+
+    character optional, intent(in) :: norm = '1'
+    integer intent(hide), depend(d) :: n = max(3, len(d))
+    <ftype> intent(in), depend(n), dimension(n - 1) :: dl
+    <ftype> intent(in), dimension(n) :: d
+    <ftype> intent(in), depend(n), dimension(n - 1) :: du
+    <ftype> intent(in), depend(n), dimension(n - 2) :: du2
+    integer intent(in), depend(n), dimension(n) :: ipiv
+    <ftypereal> intent(in) :: anorm
+    <ftypereal> intent(out) :: rcond
+    <ftype> intent(hide, cache), dimension(2*n), depend(n) :: work
+    integer intent(hide, cache), dimension(n), depend(n) :: iwork
```

Comment:
In the `gecon` wrapper, this is:
```
    <integer,integer,real, double precision> depend(n),dimension(<n,n,2*n,2*n>),intent(hide,cache):: irwork
```
Why that for an integer array? And it seems that what I've done here doesn't match what's in the `callprotoargument` section, but it still works... (Why does it?) Which should be changed?

Reply:
Because the name, as long as it is consistent, does not make difference however it depicts the fact that for complex this is a real array and not integer.